---
title: "Basics of Quantitative Genetics (Part 2)"
author: "Peter von Rohr"
date: "04 Oktober 2019"
output: beamer_presentation
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, results = "asis")
knitr::knit_hooks$set(hook_convert_odg = rmdhelp::hook_convert_odg)
```

## Variances
* selection can only be done with variation
* variation is assessed by variance

\begin{equation}
Var\left[X\right] = \sum_{x_i \in \mathcal{X}} (x_i - \mu_X)^2 * f(x_i) \notag
\end{equation}

 \vspace*{1ex}
  \begin{tabular}{p{1cm}p{1cm}p{6cm}}
  where & $\mathcal{X}$: &  set of all possible $x$-values\\
        & $f(x_i)$       &  probability that $x$ assumes the value of $x_i$ \\
        & $\mu_X $       &  expected value $E\left[X\right]$ of $X$
  \end{tabular}


## Variation of Genotypic Values

\begin{align}
\sigma_G^2 = Var\left[V\right] &=   (V_{11} - \mu)^2 * f(G_1G_1) \notag \\
                               &  +\  (V_{12} - \mu)^2 * f(G_1G_2) \notag \\
                               &  +\  (V_{22} - \mu)^2 * f(G_2G_2) \notag \\
                               &=   (BV_{11} + D_{11})^2 * f(G_1G_1) \notag \\
                               &  +\  (BV_{12} + D_{12})^2 * f(G_1G_2) \notag \\
                               &  +\  (BV_{22} + D_{22})^2 * f(G_2G_2) \notag \\
                               &=  2pq\alpha^2 + \left(2pqd \right)^2 \notag\\
                               &=  \sigma_A^2 + \sigma_D^2 \notag
\end{align}

where $\mu = (p - q)a + 2pqd$ the population mean and $V_{ij} - \mu = BV_{ij} + D_{ij}$


## Additive Genetic Variance

\begin{align}
\sigma_A^2 &= Var\left[ BV \right] =  BV_{11}^2 * f(G_1G_1) + BV_{12}^2 * f(G_1G_2) + BV_{22}^2 * f(G_2G_2) \notag \\
           &= \left(2q \alpha \right)^2 * p^2 +  \left((q-p) \alpha \right)^2 * 2pq + \left(-2p \alpha \right)^2 * q^2 \notag \\
           &= 4p^2 q^2 \alpha^2 + \left(q^2 \alpha^2 -2pq\alpha^2 + p^2 \alpha^2 \right) * 2pq + 4p^2q^2\alpha^2 \notag \\
           &= 8p^2 q^2 \alpha^2 + 2pq^3\alpha^2 -4p^2q^2\alpha^2 + 2p^3q\alpha^2 \notag \\
           &= 4p^2 q^2 \alpha^2 + 2pq^3\alpha^2  + 2p^3q\alpha^2 \notag \\
           &= 2pq\alpha^2 \left(2pq + q^2 + p^2 \right) \notag \\
           &= 2pq\alpha^2 \notag
\end{align}

because $E\left[ BV \right] = 0$


## Dominance Variance

\begin{align}
\sigma_D^2  &=  D_{11}^2 * f(G_1G_1) + D_{12}^2 * f(G_1G_2) + D_{22}^2 * f(G_2G_2) \notag \\
            &=   (- 2q^2d)^2 * p^2 + (2pqd)^2 * 2pq + (- 2p^2d)^2 * q^2 \notag \\
            &=  4p^2q^4d^2 + 8p^3q^3d^2 + 4p^4q^2d^2 \notag \\
            &=  4p^2q^2d^2 \left( q^2 + 2pq + p^2 \right) \notag \\
            &=  4p^2q^2d^2 \notag
\end{align}

because $E\left[ D \right] = 0$


## Extension To More Loci

```{r populationtwoloci, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg"}
#rmdhelp::use_odg_graphic(ps_path = "odg/populationtwoloci.odg")
knitr::include_graphics(path = "odg/populationtwoloci.png")
```
 

## Epistatic Interaction

* Decomposition of overall genotypic value $V$

\begin{equation}
V = V_A + V_B + I_{AB} \notag
\end{equation}

where $I_{AB}$:  deviation from additively combining $A$ and $B$

* Decomposition of $V_A$ and $V_B$ as before and re-grouping additive and dominance terms

\begin{equation}
V = \mu + U + D + I_{AB} \notag
\end{equation}

where $U$ : breeding values and $D$: Dominance deviations


## Interaction Variance
* Interaction deviation lead to new variance component $\sigma_I^2$

\begin{align}
\sigma_G^2 &= var(V) = var(\mu + U + D + I_{AB}) \notag \\
           &= var(U) + var(D) + var(I_{AB}) \notag \\
           &= \sigma_U^2 + \sigma_D^2 + \sigma_I^2 \notag
\end{align}



## Summary of Effects

```{r effectsummary, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg"}
#rmdhelp::use_odg_graphic(ps_path = "odg/effectsummary.odg")
knitr::include_graphics(path = "odg/effectsummary.png")
```


## Importance for Livestock Breeding
* Breeding values measure effects of alleles
* Dominance deviations measure effects of genotypes
* Interaction deviations measure effects of different loci


$\rightarrow$ for livestock breeding

* parents pass random sample of allels to offspring

$\rightarrow$ __breeding value__ and associated __genetic additive variance__ are important
 
$\rightarrow$ dominance and interaction deviations and associated variance components rarely considered


## Genetic Models

* __polygenic model__ 
    + large but finite number of loci affect phenotype 
    + used in genomic selection
* __infinitesimal model__ 
    + infinite number of loci affect phenotype
    + breeding value is the sum of infinitely small single locus breeding values
    + breeding value converges to normal distribution (Central Limit Theorem) 
    + used in traditional breeding value estimation


## Central Limit Theorem

```{r hist-clt-setup, echo=FALSE}
# fixing some constants
set.seed(9876)
n_nr_sample <- 10000
n_nr_comp <- 10^(1:3)

# define a function to compute sample means
sample_comp_sum <- function(pn_nr_comp, pn_nr_sample){
  return(sapply(1:pn_nr_comp, 
                function(x, y) sum(runif(y))/y,
                pn_nr_sample))
}
# vectors with component sums
vec_sum_10 <- sample_comp_sum(pn_nr_comp = n_nr_comp[1], pn_nr_sample = n_nr_sample)
vec_sum_100 <- sample_comp_sum(pn_nr_comp = n_nr_comp[2], pn_nr_sample = n_nr_sample)
vec_sum_1000 <- sample_comp_sum(pn_nr_comp = n_nr_comp[3], pn_nr_sample = n_nr_sample)
```

```{r hist-clt, echo=FALSE, fig.cap="Distribution of Sums of Different Numbers of Components"}
# plots
opar <- par()
par(mfrow=c(1,3))
hist(vec_sum_10)
hist(vec_sum_100)
hist(vec_sum_1000)
```



